plugins/webhooks/index.js

```diff
@@ -346,33 +375,32 @@ function pushEvent(pluginOptions, request, reply, parsed) {
             return pipeline.sync()
                 // handle the PR action
                 .then(p => p.jobs.then((jobs) => {
-                    const pipelineId = p.id;
-                    const name = 'main';
-                    const i = jobs.findIndex(j => j.name === name); // get job's index
-                    const jobId = jobs[i].id;
+                    let commitJobs = [];
+
+                    // if it's using old workflow then find the main job
+                    if (p.workflow) {
+                        commitJobs = jobs.filter(j => j.name === 'main');
+                    } else {
+                        commitJobs = jobs.filter(j => j.requires && j.requires.includes('~commit'));
```

Comment:
actually `requires` is not stored in the job object itself, so I don't think we can get the value out of `j.requires`, will probably need to get it from `permutations`

Reply:
Ah, okay, this can be ugly..